docs/cpp_docs.rst

```diff
@@ -26,94 +45,540 @@ The beginning of both header and source code files should include
+
+For internal consistency, always use ``@note`` and not the synonymous ``@remark`` or ``@remarks``.
+
+.. _cpp-doxygen-references:
+
+References
+----------
+
+References can be included either in the :ref:`'Notes' <cpp-doxygen-notes>` section or in a separate list below them. A reference consists of the ``@cite`` tag, followed by a BibTeX label. Bibfiles must be listed in the ``CITE_BIB_FILES`` configuration tag in ``doc/doxygen.conf.in``.
+
+Note that Web pages should be referenced with regular inline links.
+
+References are meant to augment the documentation, but should not be required to understand it.
+
+.. _cpp-doxygen-examples:
+
+Examples
+--------
+
+'Examples' is an optional section for examples. This section is very strongly encouraged.
+
+Examples should be prefixed by ``@code`` and suffixed by ``@endcode``:
+
+.. code-block:: cpp
+
+   /**
+    * @code
+    * auto cosines = cos(angles);
+    * @endcode
     *
-    * @author Joe Smith
-    * Contact: js@lsst.org
+    * Comment explaining the second example.
     *
+    * @code
+    * auto cosines = cos(radians(angles));
+    * @endcode
     */
 
-.. _doc-cpp-package-definition:
+.. _cpp-doxygen-package-definition:
 
-Package Documentation / Definition
-==================================
+Documenting/Defining Packages
+=============================
 
 Each LSST package corresponds to a group in Doxygen.
 We declare this package in the root header file for a package, usually named ``package.h``.
 
-In this header file, below the preamble, provide a Doxygen comment block that declares the package with the fields:
-
-1. ``@defgroup`` followed by the ``PackageName`` and  ``PackageTitle``
+In this header file, below the boilerplate but above any ``#include`` statements or other code, provide a Doxygen comment block that declares the package with the fields:
 
-2. ``@brief`` to provide a one-line description of the package.
+1. ``@defgroup`` followed by machine-readable and human-readable names for the package
+2. :ref:`cpp-doxygen-short-summary` (must be prefixed by ``@brief`` because the summary is not the first line.)
+3. :ref:`cpp-doxygen-deprecation` (if applicable)
+4. :ref:`cpp-doxygen-extended-summary` (recommended)
+5. :ref:`cpp-doxygen-see-also` (optional)
 
 For example:
 
 .. code-block:: cpp
 
    /**
-    * @defgroup PackageName PackageTitle
+    * @defgroup afw LSST Framework
+    *
+    * @brief LSST data management: astronomical framework, including images and tables
+    */
+
+Header files that belong to the package must have a Doxygen comment like
+
+.. code-block:: cpp
+
+   /**
+    * @addtogroup group
     *
-    * @brief Provide some stuff to do stuff
+    * @{
     */
 
-Class Definitions
-=================
+before any documented code, and a Doxygen comment with a ``@}`` after any documented code. Everything between the ``@{`` and the ``@}`` will be listed on the module page.
```

Comment:
No, it often did not work. For example, [the afw module documentation](https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/group__afw.html) is a list of source files rather than a list of API elements.